protocol/message.go

```diff
@@ -176,8 +164,8 @@ func NewRegistrationProof(ap *m.AuthenticationPath, str *DirSTR,
 	return &Response{
 		Error: e,
 		DirectoryResponse: &DirectoryProof{
-			AP:  ap,
-			STR: str,
+			AP:  append([]*m.AuthenticationPath{}, ap),
+			STR: append([]*DirSTR{}, str),
```

Comment:
We should update the documentation to specify that the length of `AP` and `STR` is expected to be equal to `1`. Same goes for `NewKeyLookupProof` below.